### ISSUE TITLE
Circular reference in Modelica.Media.Air.MoistAir.PsychrometricData

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1679]
**Reported by stefanv on 11 Mar 2015 17:39 UTC**
Modelica.Media.Air.MoistAir.PsychrometricData is a model containing the declaration,

```mo
package Medium = Modelica.Media.Air.MoistAir "Used medium package";
```

So, model PsychrometricData essentially contains a package that it's contained in. A tool that tries to naively walk the library will recurse forever.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1679

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 11 Mar 2015 17:56 UTC**
I believe this is a concrete case of the problem discussed in ticket #1458.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 11 Mar 2015 21:52 UTC**
I see the problem. It is also strange that an example is just there in the middle of a package of functions without motivation. I did not do that. I am not sure what the best way to solve this is that complies with rules on minor releases. Do these rules hold if the model might violate the specification? 

My simple solution would be to move this model to Media/Examples, which is where I think it belongs. Comments welcome, if there are no objections, I'll do that. And will change it in the 3.2.1 branch + trunk.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 22 Jun 2015 13:25 UTC**
Fixed in trunk in 5fd4eb90b9e1b37712cc6cab9ac918771d3ebae8

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 22 Jun 2015 14:58 UTC**
Fixed in the release branch in 84000601efc8b6900f1b42123930cb5ebe1baf74

**Comment by modelica-trac-importer** [comment]
**Changelog modified by hubertus on 22 Jun 2015 14:58 UTC**
Circular Reference bug in Modelica.Media fixed

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 25 Jul 2015 15:04 UTC**
It was forgotten to update the reference in the documentation to the new location: Fixed in e00b218a69c3b3c0881b7c96136a9b4619029bc7 and 053ed5147e40a9f5a3d4a7a3c9cd024d296df041.
